pkg/analysis/passes/license/license.go

```diff
@@ -0,0 +1,61 @@
+package license
+
+import (
+	"strings"
+
+	"github.com/go-enry/go-license-detector/v4/licensedb"
+	"github.com/go-enry/go-license-detector/v4/licensedb/filer"
+	"github.com/grafana/plugin-validator/pkg/analysis"
+	"github.com/grafana/plugin-validator/pkg/analysis/passes/archive"
+)
+
+var (
+	licenseNotProvided = &analysis.Rule{Name: "license-not-provided", Severity: analysis.Error}
+)
+
+var Analyzer = &analysis.Analyzer{
+	Name:     "license",
+	Requires: []*analysis.Analyzer{archive.Analyzer},
+	Run:      run,
+	Rules:    []*analysis.Rule{licenseNotProvided},
+}
+
+var validLicenseStart = []string{"AGPL-3.0", "Apache-2.0"}
+
+const minRequiredConfidenceLevel float32 = 0.9
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	archiveDir := pass.ResultOf[archive.Analyzer].(string)
+
+	filer, err := filer.FromDirectory(archiveDir)
+	if err != nil {
+		pass.ReportResult(pass.AnalyzerName, licenseNotProvided, "License not found", "Could not find or parse the license file inside the plugin archive. Please make sure to include a LICENCE file in your archive.")
+		return nil, nil
+	}
+
+	licenses, err := licensedb.Detect(filer)
+	if err != nil {
+		pass.ReportResult(pass.AnalyzerName, licenseNotProvided, "License not found", "Could not find or parse the license file inside the plugin archive. Please make sure to include a LICENCE file in your archive.")
```

Comment:
I am reporting this and the following as errors instead of returning the error. Returning the error will stop the whole plugincheck process to stop checking.